src/get-new-changes.test.ts

```diff
@@ -0,0 +1,451 @@
+
+      // Run 1: Initial commits
+      mockRunCommandAndSplit.mockResolvedValueOnce(['commit1', 'commit2']);
+
+      mockRunCommand
+        .mockResolvedValueOnce('feat: feature (#12345)')
+        .mockResolvedValueOnce('Update Attributions'); // Direct commit
+
+      const result1 = await getNewChangeEntries({
+        mostRecentTag: 'v1.0.0',
+        repoUrl,
+        loggedPrNumbers: [],
+        loggedDescriptions: [],
+        useChangelogEntry: false,
+        useShortPrLink: false,
+      });
+
+      // First run: Both should be included
+      expect(result1).toHaveLength(2);
+      expect(result1[0].description).toContain('12345');
+      expect(result1[1].description).toBe('Update Attributions');
+
+      // Run 2: Same commits + new one (simulating re-run after new push)
+      jest.clearAllMocks();
+      mockRunCommandAndSplit.mockResolvedValueOnce([
+        'commit1',
+        'commit2',
+        'commit3',
+      ]);
+
+      mockRunCommand
+        .mockResolvedValueOnce('feat: feature (#12345)') // Already logged
+        .mockResolvedValueOnce('Update Attributions') // Already logged - KEY FIX
+        .mockResolvedValueOnce('fix: new fix (#12346)'); // New commit
+
+      const result2 = await getNewChangeEntries({
+        mostRecentTag: 'v1.0.0',
+        repoUrl,
+        loggedPrNumbers: ['12345'], // First PR now logged in changelog
+        loggedDescriptions: ['feat: feature', 'Update Attributions'], // Both now logged (PR link stripped)
+        useChangelogEntry: false,
+        useShortPrLink: false,
+      });
+
+      // Should only include the new PR, NOT the duplicates
+      expect(result2).toHaveLength(1);
+      expect(result2[0].description).toContain('12346');
+      expect(
+        result2.find((item) => item.description.includes('12345')),
+      ).toBeUndefined();
+      expect(
+        result2.find((item) => item.description.includes('Attributions')),
+      ).toBeUndefined(); // CRITICAL: No duplicate of "Update Attributions"
+    });
+
+    it('should handle all commits being duplicates', async () => {
+      mockRunCommandAndSplit.mockResolvedValueOnce([
+        'commit1',
+        'commit2',
+        'commit3',
+      ]);
+
+      mockRunCommand
+        .mockResolvedValueOnce('feat: feature (#12345)') // Already logged
+        .mockResolvedValueOnce('Update Attributions') // Already logged
+        .mockResolvedValueOnce('Bump version'); // Already logged
+
+      const result = await getNewChangeEntries({
+        mostRecentTag: 'v1.0.0',
+        repoUrl,
+        loggedPrNumbers: ['12345'],
+        loggedDescriptions: ['Update Attributions', 'Bump version'],
+        useChangelogEntry: false,
+        useShortPrLink: false,
+      });
+
+      // Should return empty array - all duplicates
+      expect(result).toHaveLength(0);
+    });
+  });
+
+  describe('edge cases', () => {
+    it('should handle no commits since last tag', async () => {
```

Comment:
How should it "handle" no commits? It would be better to be more clear about the result:

``` suggestion
    it('should return no entries if there are no commits', async () => {